graph/patterns/upsert.md

```diff
@@ -0,0 +1,173 @@
+# Upsert
+
+Microsoft Graph API Design Pattern
+
+*The `UPSERT` pattern is a non-destructive idempotent operation using a client-provided key, that ensures that system resources can be deployed in a reliable, repeatable, and controlled way, typically used in Infrastructure as Code (IaC) scenarios.*
+
+## Problem
+
+Infrastructure as code (IaC) defines system resources and topologies in a descriptive manner that allows teams to manage those resources as they would code.
```

Comment:
Should we say that Infrastructure as code (IaC) defines system resources and topologies in a **declarative** manner. It's an industry term i believe.

Reply:
Happy to make that change.  I took most of this text straight from Azure docs :)